datatypes/src/primitives/bounding_box.rs

```diff
@@ -386,6 +386,39 @@ impl BoundingBox2D {
             None
         }
     }
+
+    pub fn add_coord(&mut self, coord: Coordinate2D) {
+        self.lower_left_coordinate = self.lower_left_coordinate.min_elements(coord);
+        self.upper_right_coordinate = self.upper_right_coordinate.max_elements(coord);
+    }
+
+    pub fn from_coord_iter<I: IntoIterator<Item = Coordinate2D>>(iter: I) -> Option<Self> {
+        let mut iterator = iter.into_iter();
+
+        let first = iterator.next().map(|c| BoundingBox2D::new_unchecked(c, c));
+
+        first.map(|mut f| {
+            for c in iterator {
+                f.add_coord(c);
+            }
+            f
+        })
+    }
```

Comment:
I guess you can call `from_coord_ref_iter` by using `as_ref` on `iter`.

Reply:
i don't think that is possible. who would own the refs? I can make the ref one use cloned points...

Reply:
You are right, we could call `copied` or we can use the `Borrow` trait.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9e52d6b3f20de4f1365c209b071ec551

It is up to you if you want to look into it.

Reply:
i would say we keep it as it is and revisit this later.